app/services/cache/mapping/declaration/DeclarationBuilder.scala

```diff
@@ -0,0 +1,91 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.cache.mapping.declaration
+
+import forms.declaration.RepresentativeDetails
+import services.cache.ExportsCacheModel
+import services.mapping.declaration._
+import wco.datamodel.wco.dec_dms._2.Declaration
+import wco.datamodel.wco.declaration_ds.dms._2.{DeclarationFunctionalReferenceIDType, DeclarationTypeCodeType}
+
+object DeclarationBuilder {
+
+  val defaultFunctionCode = "9"
+
+  def build(exportsCacheModel: ExportsCacheModel): Declaration = {
+    val declaration = new Declaration()
+
+    declaration.setFunctionCode(FunctionCodeBuilder.build(defaultFunctionCode))
+    declaration.setFunctionalReferenceID(createDeclarationFunctionalReferenceIDType(exportsCacheModel))
+
+    val typeCode: Option[DeclarationTypeCodeType] = exportsCacheModel.additionalDeclarationType.map(
+      additionalDeclarationType =>
+        TypeCodeBuilder.createTypeCode(additionalDeclarationType, exportsCacheModel.dispatchLocation)
+    )
+
+    declaration.setTypeCode(typeCode.getOrElse(new DeclarationTypeCodeType()))
+    declaration.setGoodsItemQuantity(GoodsItemQuantityBuilder.createGoodsItemQuantity(exportsCacheModel.items.toSeq))
+
+    generateAgent(exportsCacheModel).foreach { data =>
+      declaration.setAgent(data)
+    }
+
+    //    declaration.setGoodsShipment(GoodsShipmentBuilder.build)
+    //    declaration.setExitOffice(ExitOfficeBuilder.build)
+    //    declaration.setBorderTransportMeans(BorderTransportMeansBuilder.build)
+    //    declaration.setExporter(ExporterBuilder.build)
+    //    declaration.setDeclarant(DeclarantBuilder.build)
+    //    declaration.setInvoiceAmount(InvoiceAmountBuilder.build)
+    //    declaration.setPresentationOffice(PresentationOfficeBuilder.build)
+    //    declaration.setSpecificCircumstancesCodeCode(SpecificCircumstancesCodeBuilder.build)
+    //    declaration.setSupervisingOffice(SupervisingOfficeBuilder.build)
+    //    declaration.setTotalPackageQuantity(TotalPackageQuantityBuilder.build)
+    //    declaration.setConsignment(DeclarationConsignmentBuilder.build)
+    //    declaration.setTypeCode(TypeCodeBuilder.build)
+    //
+    //    val authorisationHolders = AuthorisationHoldersBuilder.build
+    //    if (authorisationHolders != null && !authorisationHolders.isEmpty) {
+    //      declaration.getAuthorisationHolder.addAll(authorisationHolders)
+    //    }
+    //
+    //    val currencyExchangeList = CurrencyExchangeBuilder.build
+    //    if (currencyExchangeList != null && !currencyExchangeList.isEmpty) {
+    //      declaration.getCurrencyExchange.addAll(currencyExchangeList)
+    //    }
+
+    declaration
+  }
+
+  private def createDeclarationFunctionalReferenceIDType(
```

Comment:
This is still early days, but are you thinking of declaring all the functions to map a DMS declaration?
This will turn this file into a quite big file.